enterprise/server/backends/pebble_cache/pebble_cache.go

```diff
@@ -95,11 +95,18 @@ const (
 // Options is a struct containing the pebble cache configuration options.
 // Once a cache is created, the options may not be changed.
 type Options struct {
-	RootDirectory       string
-	Partitions          []disk.Partition
-	PartitionMappings   []disk.PartitionMapping
-	MaxSizeBytes        int64
-	BlockCacheSizeBytes int64
+	RootDirectory     string
+	Partitions        []disk.Partition
+	PartitionMappings []disk.PartitionMapping
+
+	MaxSizeBytes           int64
+	BlockCacheSizeBytes    int64
+	MaxInlineFileSizeBytes int64
+
+	AtimeUpdateThreshold time.Duration
+	AtimeWriteBatchSize  int
+	AtimeBufferSize      int
+	MinEvictionAge       time.Duration
```

Comment:
I worry a little about partially filled options here -- wdyt?

Some options for fixing might be:
 - move the flag default values to consts, set anything in options to its proper default value if it's not set
 - leave it, just write a big scary comment
 - add some options validation code that barfs if it's a bad config

Reply:
Good call out - done

Reply:
Made options struct private and two constructors to make it explicit that there are two ways to create options - either everything gets set via flags, or non-nil fields are grabbed from the OptionsConstructor.

I added the OptionsConstructor bc it's valid for certain fields, like AtimeUpdateThreshold, to be 0, so we need pointers to differentiate whether a 0 value was explicitly set by the user, vs was just set as a default value